src/Symfony/Component/Notifier/Bridge/Sinch/SinchTransportFactory.php

```diff
@@ -29,6 +30,11 @@ public function create(Dsn $dsn): TransportInterface
         $accountSid = $this->getUser($dsn);
         $authToken = $this->getPassword($dsn);
         $from = $dsn->getOption('from');
+
+        if (!$from) {
+            throw new IncompleteDsnException('Missing from.', $dsn->getOriginalDsn());
+        }
```

Comment:
This should be considered a bugfix

Reply:
I agree. This case resulted in a `TypeError` previously, so `IncompleteDsnException` is the more accurate exception. However, I think we pull up the scheme check in this case.

Reply:
You are right, the best thing would be to move the scheme check up and check that first before checking for required parameters.

1. Check if the `scheme` is supported, if not -> `UnsupportedSchemeException`
2. If `scheme` is supported, but not all requirements are fulfilled -> `IncompleteDsnException`

Right now you could get an `IncompleteDsnException` before even the schema is validated.

Proposal (for all transports):
```diff
     public function create(Dsn $dsn): TransportInterface
     {
         $scheme = $dsn->getScheme();
+
+        if ('freemobile' !== $scheme) {
+            throw new UnsupportedSchemeException($dsn, 'freemobile', $this->getSupportedSchemes());
+        }
+
         $login = $this->getUser($dsn);
         $password = $this->getPassword($dsn);
         $phone = $dsn->getOption('phone');

         if (!$phone) {
             throw new IncompleteDsnException('Missing phone.', $dsn->getOriginalDsn());
         }

-        if ('freemobile' === $scheme) {
-            return new FreeMobileTransport($login, $password, $phone, $this->client, $this->dispatcher);
-        }
-
-        throw new UnsupportedSchemeException($dsn, 'freemobile', $this->getSupportedSchemes());
+        return new FreeMobileTransport($login, $password, $phone, $this->client, $this->dispatcher);
     }
```

WDYT @derrabus @fabpot and should this be considered a bugfix starting from `5.1` or a new feature?

Reply:
Currently on 5.1, when I use an unsupported DSN (for instance some HTTPS URL), I would get an `UnsupportedSchemeException`. And this is the exception that I would expect in that case. With this PR in its current state, this would be turned into a `IncompleteDsnException` (because an HTTPS URL does not have a `from` option).

Imho, the scheme check has to be pulled to the top of the method (as you've suggested, @OskarStark) in order to not break the current behavior.

Reply:
Makes sense 👍  so bugfix

Reply:
Done @derrabus